ground/gcs/src/plugins/config/configautotunewidget.cpp

```diff
@@ -407,23 +395,309 @@ void ConfigAutotuneWidget::stuffShareForm(AutotuneFinalPage *autotuneShareForm)
     }
 }
 
+// XXX appropriately encapsulate all this crap
+#define ATFLASH_MAGIC 0x656e755480008041 /* A...Tune */
+struct at_flash_header
+{
+    uint64_t magic;
+    uint16_t wiggle_points;
+    uint16_t aux_data_len;
+    uint16_t sample_rate;
+
+    // Consider total number of averages here
+    uint16_t resv;
+};
+
+struct at_measurement
+{
+    float y[3]; /* Gyro measurements */
+    float u[3]; /* Actuator desired */
+};
+
+struct at_flash
+{
+    struct at_flash_header hdr;
+
+    struct at_measurement data[];
+};
+
+/* Run a Butterworth biquad filter on a circular buffer.  First go around once
+ * to "prime" the filter, then actually filter in place.  This is derived from
+ * @glowtape's excellent flight implementation
+ */
+void biquad_filter_in_place(float cutoff, int pts, float *data)
```

Comment:
Plans to encapsulate it in classes.. I'm not sure about how much further to go.

Kinda **regret** doing it for the other stuff, rather than just doing it c style ,given that this code may need to get factored back to flight in the future.